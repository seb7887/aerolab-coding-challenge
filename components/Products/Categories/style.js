import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 1.5rem;
  display: flex;
  flex-direction: column;

  h3 {
    color: ${props => props.theme.secondary};
  }

  a {
    margin-bottom: 1rem;
    text-transform: capitalize;
    font-size: 1rem;
    transition: all 0.5s;

    &.current {
      font-weight: bold;
      text-decoration: underline;
    }

    &:hover {
      text-decoration: underline;
    }
  }

  @media (${props => props.theme.sm}) {
    display: none;
  }
`;
