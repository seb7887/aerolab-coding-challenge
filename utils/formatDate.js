export default function formatDate(value) {
  const getNumber = number => (number < 10 ? `0${number}` : String(number));
  const date = new Date(value);

  return `${getNumber(date.getMonth() + 1)}/${getNumber(
    date.getDate()
  )}/${getNumber(date.getFullYear())} ${getNumber(
    date.getUTCHours()
  )}:${getNumber(date.getUTCMinutes())}`;
}
