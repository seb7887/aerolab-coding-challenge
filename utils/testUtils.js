import faker from 'faker';

export const fakeUser = {
  name: 'John Kite',
  points: '7000'
};

export const mockCategories = [
  'all',
  'gaming',
  'laptops',
  'tablets',
  'cameras',
  'audio',
  'tv',
  'drones',
  'accessories',
  'home'
];

export const fakeProduct = {
  _id: 7,
  name: 'iPhone',
  category: 'Phones',
  cost: 800,
  img: {
    url: '/static/image/image.png'
  }
};

export const generateProduct = () => {
  const newProduct = {
    _id: faker.random.uuid(),
    name: faker.commerce.productName(),
    category: faker.commerce.productAdjective(),
    cost: 800,
    img: {
      url: faker.image.imageUrl()
    }
  };
  return newProduct;
};

export const fakeProducts = [fakeProduct];

export const generateProducts = quantity => {
  let products = [];

  for (let i = 0; i < quantity; i++) {
    const newProduct = generateProduct();
    products.push(newProduct);
  }

  return products;
};
