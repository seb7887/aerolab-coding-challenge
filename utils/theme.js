const black = '#000';
const white = '#fff';
const cyan = '#0ad4fa';
const orange = '#ff6e05';
const red = 'red';
const green = 'green';

const bgLight = '#e9ebf1';
const bgDark = '#212531';
const secondaryBgLigth = 'rgba(0, 0, 0, 0.1)';
const secondaryBgDark = 'rgba(255, 255, 255, 0.1)';
const containerBgLight = '#fff';
const containerBgDark = '#000';
const boxShadowLight = '0px 0px 3px 0px rgba(0, 0, 0, 0.1)';
const boxShadowDark = 'rgba(31, 31, 31, 0.1)';
const productShadowLight = '1px 4px 5px 1px rgba(0, 0, 0, 0.1)';
const productShadowDark = '1px 4px 5px 1px rgba(255, 255, 255, 0.1)';

const primaryLight = '#616161';
const primaryDark = '#f8f8f8';

const secondaryLight = black;
const secondaryDark = white;
const greyLight = '#e0e0e0';
const greyDark = '#a3a3a3';

const theme = {
  light: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyLight,
    bg: bgLight,
    secondaryBg: secondaryBgLigth,
    containerBg: containerBgLight,
    boxShadow: boxShadowLight,
    productShadow: productShadowLight,
    primary: primaryLight,
    secondary: secondaryLight
  },
  dark: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyDark,
    bg: bgDark,
    secondaryBg: secondaryBgDark,
    containerBg: containerBgDark,
    boxShadow: boxShadowDark,
    productShadow: productShadowDark,
    primary: primaryDark,
    secondary: secondaryDark
  }
};

export default theme;
