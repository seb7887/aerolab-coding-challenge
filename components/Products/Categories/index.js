import React from 'react';
import Link from 'next/link';
import shortid from 'shortid';

import { Container } from './style';

const categories = [
  'all',
  'gaming',
  'laptops',
  'tablets',
  'cameras',
  'audio',
  'tv',
  'drones',
  'accessories',
  'home'
];

const Categories = ({ category, sort }) => (
  <Container>
    <h3>Categories</h3>
    <>
      {categories.map(name => {
        const isCurrent = name === category;
        return (
          <Link
            prefetch
            href={{
              pathname: '/',
              query: {
                category: name,
                sort,
                page: 0
              }
            }}
            key={shortid.generate()}
          >
            <a className={isCurrent ? 'current' : ''}>{name}</a>
          </Link>
        );
      })}
    </>
  </Container>
);

export default Categories;
