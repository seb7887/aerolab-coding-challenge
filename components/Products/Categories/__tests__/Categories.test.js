import { render } from 'react-testing-library';

import Categories from '../index';

import { mockCategories } from '../../../../utils/testUtils';

jest.mock('next/link', () => {
  return ({ children }) => {
    return children;
  };
});

const category = 'all';
const sort = 'recent';

describe('Categories', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <Categories category={category} sort={sort} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display Categories title', () => {
    const { getByText } = render(
      <Categories category={category} sort={sort} />
    );

    expect(getByText(/categories/i)).toBeTruthy();
  });

  it('should display all the categories', () => {
    const { getAllByTestId } = render(
      <Categories category={category} sort={sort} />
    );

    expect(getAllByTestId('category').length).toBe(mockCategories.length);
  });
});
