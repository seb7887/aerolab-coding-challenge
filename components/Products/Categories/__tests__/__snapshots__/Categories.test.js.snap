// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__Container-sc-15ofmo2-0 SyiiL"
  >
    <h3>
      Categories
    </h3>
    <a
      class="current"
      data-testid="category"
    >
      all
    </a>
    <a
      class=""
      data-testid="category"
    >
      gaming
    </a>
    <a
      class=""
      data-testid="category"
    >
      laptops
    </a>
    <a
      class=""
      data-testid="category"
    >
      tablets
    </a>
    <a
      class=""
      data-testid="category"
    >
      cameras
    </a>
    <a
      class=""
      data-testid="category"
    >
      audio
    </a>
    <a
      class=""
      data-testid="category"
    >
      tv
    </a>
    <a
      class=""
      data-testid="category"
    >
      drones
    </a>
    <a
      class=""
      data-testid="category"
    >
      accessories
    </a>
    <a
      class=""
      data-testid="category"
    >
      home
    </a>
  </div>
</DocumentFragment>
`;
